### ISSUE TITLE
DHCP server not working in Freebsd

### ISSUE THREAD

**Issue by aserrallerios** [AdguardTeam/AdGuardHome#3967]
Version: 0.107.0.b.7

I try to configure and run the DHCP server in Freebsd, when I click "Enable DHCP server" I get this error:

> Error: control/dhcp/set_config | enabling dhcp: setting static ip: setting static ip is unsupported on freebsd | 500

The server looks like enabled (I can see the "Disable DHCP server" where previously I had the "Enable...") but it's not running, as I try to get a new IP from a new host and it doesn't work.

Probably [this fix](https://github.com/AdguardTeam/AdGuardHome/issues/686) is not working in Freebsd?

**Comment by aserrallerios** [comment]
I've found this issue: https://github.com/AdguardTeam/AdGuardHome/issues/3289

So it should work, but somehow it doesn't for me. Let me double validate that the DHCP is not working.

**Comment by aserrallerios** [comment]
I cannot make it work. I'm checking the logs and I can only see this:

```
2021/12/20 22:27:52.620549 [info] AdGuard Home is available at the following addresses:
2021/12/20 22:27:52.620766 [info] Go to http://[::1]:80
2021/12/20 22:27:52.622168 [info] Go to http://[::1]:3001 (BETA)
2021/12/20 22:27:52.622910 [info] Go to http://127.0.0.1:80
2021/12/20 22:27:52.625100 [info] Go to http://127.0.0.1:3001 (BETA)
2021/12/20 22:27:52.625157 [info] Go to http://192.168.1.6:80
2021/12/20 22:27:52.625463 [info] Go to http://192.168.1.6:3001 (BETA)
2021/12/20 22:27:52.801735 [info] Starting the DNS proxy server
2021/12/20 22:27:52.801779 [info] Ratelimit is enabled and set to 20 rps
2021/12/20 22:27:52.801792 [info] The server is configured to refuse ANY requests
2021/12/20 22:27:52.801804 [info] DNS cache is enabled
2021/12/20 22:27:52.801819 [info] MaxGoroutines is set to 300
2021/12/20 22:27:52.801835 [info] Creating the UDP server socket
2021/12/20 22:27:52.801884 [info] Listening to udp://[::]:53
2021/12/20 22:27:52.801926 [info] Creating a TCP server socket
2021/12/20 22:27:52.801962 [info] Listening to tcp://[::]:53
2021/12/20 22:27:52.804310 [info] Entering the UDP listener loop on [::]:53
2021/12/20 22:27:52.804375 [info] Entering the tcp listener loop on [::]:53
2021/12/20 22:42:04.690884 [info] DHCP: POST /control/dhcp/set_config: enabling dhcp: setting static ip: setting static ip is unsupported on freebsd
2021/12/20 22:42:05.398421 [info] DHCP: POST /control/dhcp/set_config: enabling dhcp: setting static ip: setting static ip is unsupported on freebsd
2021/12/20 22:54:13.684969 [info] DHCP: POST /control/dhcp/set_config: enabling dhcp: setting static ip: setting static ip is unsupported on freebsd
2021/12/20 22:57:45.641001 [info] DHCP: POST /control/dhcp/set_config: enabling dhcp: setting static ip: setting static ip is unsupported on freebsd
2021/12/21 10:17:50.567026 [info] DHCP: POST /control/dhcp/set_config: enabling dhcp: setting static ip: setting static ip is unsupported on freebsd
```

**Comment by TheCeet** [comment]
There are several fixes on DHCP in 0.107 beta cycle, try updating to latest beta?

**Comment by EugeneOne1** [comment]
@aserrallerios, the thing is setting static IP address is currently not supported on FreeBSD.  Since static address is required for DHCP server to work, AdGuard Home tries to set it anyway and fails at this step.  Could you please also try to set static IP configuration manually, e.g. like described [here](https://www.freebsd.org/cgi/man.cgi?rc.conf(5)) in the `network_interfaces` section.

**Comment by aserrallerios** [comment]
Thanks guys. 

I'm super noob so I'd appreciate any help. This is my `ifconfig` output:

```
...
epair0b: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> metric 0 mtu 1500
	options=8<VLAN_MTU>
	ether 06:d4:c4:9c:5a:ff
	hwaddr 02:07:3f:3e:1d:0b
	inet 192.168.1.6 netmask 0xffffff00 broadcast 192.168.1.255
	groups: epair
	media: Ethernet 10Gbase-T (10Gbase-T <full-duplex>)
	status: active
	nd6 options=1<PERFORMNUD>
...
```

I have the DHCP client disabled in this host. The `192.168.1.6` IP is static.

Can you tell me what to put in the `rc.conf` file?

**Comment by EugeneOne1** [comment]
Your `rc.conf` file should contain something like:

```none
ifconfig_epair0b="inet 192.168.1.6 netmask 0xffffff00"
```

Also make sure all the other lines containing "`epair0b`" are commented out with `#`.

**Comment by aserrallerios** [comment]
It worked! The UI is a bit buggy (as the "check" still doesn't work) but finally I could send the "enable" request and it worked.

Many thanks!

**Comment by aserrallerios** [comment]
Something wrong is happening now, coincidence? 

```
❯ nslookup google.com 192.168.1.6                                                                                                                                     at 
Server:		192.168.1.6
Address:	192.168.1.6#53

** server can't find google.com: SERVFAIL
```

**Comment by EugeneOne1** [comment]
@aserrallerios, could you verify that the network is reachable from AGH's host machine after interface's configuration changed?  For example, you may `ping 8.8.8.8`.

Also, how have you conclude the `192.168.1.6` is static?  Are you sure there are no other DHCP servers in the network?

**Comment by aserrallerios** [comment]
You're right, no network from AGH.

> Also, how have you conclude the 192.168.1.6 is static? Are you sure there are no other DHCP servers in the network?

Good question. The AGH host is a TrueNAS jail configured to have an static address:

<img width="944" alt="Screenshot 2021-12-21 at 16 22 59" src="https://user-images.githubusercontent.com/3159593/146954909-9a387009-90cb-4cf6-912a-e0005d0d0ceb.png">

So it looks like the jail config conflicts somehow with the `ifconfig_epair0b="inet 192.168.1.6 netmask 0xffffff00"` option. So there's any other possible workaround to trick AGH to believe it has an static address?

**Comment by aserrallerios** [comment]
Fixed it adding `defaultrouter="192.168.1.1"` to the `rc.conf`!
